src/back/glsl/mod.rs

```diff
@@ -1381,21 +1393,25 @@ impl<'a, W: Write> Writer<'a, W> {
                 }
                 write!(self.out, ")")?;
 
-                match level {
-                    // Auto needs no more arguments
-                    crate::SampleLevel::Auto => (),
-                    // Zero needs level set to 0
-                    crate::SampleLevel::Zero => write!(self.out, ", 0")?,
-                    // Exact and bias require another argument
-                    crate::SampleLevel::Exact(expr) | crate::SampleLevel::Bias(expr) => {
-                        write!(self.out, ", ")?;
-                        self.write_expr(expr, ctx)?;
-                    }
-                    crate::SampleLevel::Gradient { x, y } => {
-                        write!(self.out, ", ")?;
-                        self.write_expr(x, ctx)?;
-                        write!(self.out, ", ")?;
-                        self.write_expr(y, ctx)?;
+                if workaround_lod_array_shadow_as_grad {
```

Comment:
sorry, this is incorrect. `workaround_lod_array_shadow_as_grad` is only valid for `SampleLevel::Zero` and `SampleLevel::Exact` variants

Reply:
Thank you! Should be fixed now 😄